numpy/core/src/multiarray/arrayobject.c

```diff
@@ -1769,6 +1770,79 @@ array_free(PyObject * v)
     PyObject_Free(v);
 }
 
+static const char *
+_attr_name_as_utf8(PyObject *attr_name)
+{
+    const char *c_attr_name;
+    if (PyUString_Check(attr_name)) {
+        c_attr_name = PyUString_AsUTF8(attr_name);
```

Comment:
here and below, just `return PyUString_AsUTF8(attr_name);`? After all, you just want to return `NULL` in case this is `NULL`...